AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -25,9 +25,11 @@ final class MicrophoneManager: NSObject, ObservableObject {
     private var levelTimer: Timer?
     private(set) var session: Session?
     private lazy var locationProvider = LocationProvider()
+    let sessionSynchronizer: SessionSynchronizer
```

Comment:
Good thing you used a `protocol` here, but overall I think this class is a huge dump of logic that should really be handled elsewhere. It's usually the case with objects named `Manager` ;) Let's leave it like this for now, but in the future try to find solutions that doesn't touch those classes :)